### PR TITLE
add post content replacer (WP-671)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**

### CODE REVIEW

File: inc/Smartling/Helpers/GutenbergBlockHelper.php [hunk 1]
```diff
@@ -89,7 +106,6 @@ public function processAttributes(?string $blockName, array $flatAttributes): ar
 
             $this->getLogger()->debug(vsprintf('Pre filtered block \'%s\' attributes \'%s\'',
                 [$blockName, $ve_attributes]));
-            $this->postReceiveFiltering($flatAttributes);
```
Comment (vsolovei-smartling) [inline inc/Smartling/Helpers/GutenbergBlockHelper.php:92]: Function result unused, probably safe to delete

**Comment by vsolovei-smartling** [comment]
Obviously, tests are needed

File: inc/Smartling/Helpers/GutenbergBlockHelper.php [hunk 2]
```diff
@@ -76,6 +81,52 @@ public function register(): void
         }
     }
 
+    public function replacePreTranslateBlockContent(GutenbergBlock $block): GutenbergBlock
+    {
+        foreach ($block->getInnerBlocks() as $index => $innerBlock) {
+            $block = $block->withInnerBlock($this->replacePreTranslateBlockContent($innerBlock), $index);
```
Comment (PavelLoparev) [inline inc/Smartling/Helpers/GutenbergBlockHelper.php:87]: Do we need to stop recursion at some point manually? I mean do we need some max deep level?

**Review by PavelLoparev — APPROVED** [review]

File: inc/Smartling/Helpers/GutenbergBlockHelper.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/GutenbergBlockHelper.php:92]: Integration tests cover this

File: inc/Smartling/Helpers/GutenbergBlockHelper.php (hunk 2 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Helpers/GutenbergBlockHelper.php:87]: While we might run into issues if nesting is absurdly high, it's unlikely

**Pull request merged**